contracts/twap/CumulativeTwap.sol

```diff
@@ -101,8 +101,8 @@ contract CumulativeTwap is BlockContext {
         }
         // case3. in the middle
         else {
-            uint256 observationTimeDelta = atOrAfter.timestamp - beforeOrAt.timestamp;
-            uint256 targetTimeDelta = targetTimestamp - beforeOrAt.timestamp;
+            uint256 observationTimeDelta = atOrAfter.timestamp.sub(beforeOrAt.timestamp);
```

Comment:
we check underflow on `CumulativeTwap:157`, but still use safeMath to handle math operation